src/hybrid_crypto/header.rs

```diff
@@ -39,7 +46,12 @@ impl Metadata {
         let sec_len = scanner.read_u32()? as usize;
         let sec = scanner.next(sec_len)?.to_vec();
         let additional_data_len = scanner.read_u32()? as usize;
-        let additional_data = scanner.next(additional_data_len)?.to_vec();
+
+        let mut additional_data = None;
+        if additional_data_len > 0 {
+            additional_data = Some(scanner.next(additional_data_len)?.to_vec());
+        }
```

Comment:
do the `if {} else {}` statement directly at the initialization. It allows us to keep `additional_data` immutable.
```suggestion
        let additional_data = if additional_data_len > 0 {
            Some(scanner.next(additional_data_len)?.to_vec())
        } else {
            None
        };
```